src/main/java/graphql/schema/idl/SchemaTypeDirectivesChecker.java

```diff
@@ -160,8 +196,219 @@ private boolean isNoNullArgWithoutDefaultValue(InputValueDefinition definitionAr
         return definitionArgument.getType() instanceof NonNullType && definitionArgument.getDefaultValue() == null;
     }
 
-    private boolean argValueMatchesAllowedType(TypeDefinitionRegistry typeRegistry, Value instanceValue, Type allowedArgType) {
-        // this code is not written yet and is tricky.
-        return true;
+
+    /**
+     * Class to check whether a given directive argument value
+     * matches a given directive definition.
+     *
+     */
+    private class ArgValueOfAllowedTypeChecker {
```

Comment:
I would move this out into its own top level class.  These files are already large enough.

Reply:
Good idea, will do.